src/lib/pubkey/dilithium/dilithium_common/dilithium_keys.h

```diff
@@ -0,0 +1,114 @@
+/*
+ * Crystals Dilithium Internal Key Types
+ *
+ * (C) 2021-2024 Jack Lloyd
+ *     2021-2023 Jack Lloyd
+ *     2021-2022 Manuel Glaser - Rohde & Schwarz Cybersecurity
+ *     2021-2023 Michael Boric, René Meusel - Rohde & Schwarz Cybersecurity
+ *     2024      René Meusel - Rohde & Schwarz Cybersecurity
+ *
+ * Botan is released under the Simplified BSD License (see license.txt)
+ */
+
+#ifndef BOTAN_DILITHIUM_INTERNAL_KEYS_H_
+#define BOTAN_DILITHIUM_INTERNAL_KEYS_H_
+
+#include <botan/internal/dilithium_types.h>
+
+#include <botan/internal/dilithium_algos.h>
+#include <botan/internal/dilithium_symmetric_primitives.h>
+
+namespace Botan {
+
+class Dilithium_Keypair_Codec {
+   public:
+      static std::unique_ptr<Dilithium_Keypair_Codec> create(DilithiumMode mode);
+
+      virtual ~Dilithium_Keypair_Codec() = default;
+      virtual secure_vector<uint8_t> encode_keypair(DilithiumInternalKeypair keypair) const = 0;
+      virtual DilithiumInternalKeypair decode_keypair(std::span<const uint8_t> private_key,
```

Comment:
Mhh, they do need/generate both the public and private keys, because the public key is (partially) serialized into the private key (in Kyber/Dilithium). Hence the name.